src/linker/Linker/TypeMapInfo.cs

```diff
@@ -407,6 +410,39 @@ static bool TypeMatch (GenericParameter a, GenericParameter b)
 			return true;
 		}
 
+		static bool TypeMatch (FunctionPointerType a, FunctionPointerType b)
```

Comment:
Not 100% this is covering everything, but it at least appears to work correctly with the function pointers I have declared.

Reply:
Most notably, function pointers can't be generic but they can have generic arguments.

There isn't a `context` available here, so I'm unsure if there needs to be any additional handling or logic to cover that scenario.

Reply:
There are also `modreq` that can appear as part of the calling convention. I think `modreq` are handled further up in the `TypeMatch` logic, however.